store/mongo/datastore_mongo.go

```diff
@@ -368,22 +375,47 @@ func (db *DataStoreMongo) upsertAttributes(
 	case 1:
 		var res *mongo.UpdateResult
 		if withRevision {
-			filter = bson.M{
-				"_id":         devices[0].Id,
-				DbDevRevision: bson.M{"$lt": devices[0].Revision},
+			if etag != nil && *etag != "" {
+				filter = bson.M{
```

Comment:
I think it should be possible to build filter in a different way to avoid nested checks
e.g. https://github.com/mendersoftware/deployments/blob/master/store/mongo/datastore_mongo.go#L1794

Reply:
Yes. Good point. Thanks!